### DISCUSSION TITLE
No TS error highlight when not passing required field

### DISCUSSION THREAD

**Discussion by wasya243** [typegoose/typegoose#820] [Q&A]
I have a question about the following case, can you please help? Let's say we have the following model

```ts
import { prop } from '@typegoose/typegoose';

class Company {
    @prop({ required: true })
    name!: string;

    @prop()
    address?: string;
}

const CompanyModel = getModelForClass(Company, {
    schemaOptions: {
        toJSON: { virtuals: true },
        toObject: { virtuals: true }
      }
});

async function run() {
    try {
        const company = await CompanyModel.create({
            address: 'some address'
        });
        
        process.exit(0);
    } catch(err) {
        console.error(err);
        process.exit(1);
    }
}
```

You see that name is a required field. If I did not pass it into .create function I do not get any warnings from IDEA. If I run this code I indeed get error that name field is required and bla bla bla. Is it possible so that IDEA warns you about such cases so you do not have to run code in order to get understanding of which required fields are missing? It gets increasingly annoying when working with way bigger models with nesting.

**Comment by hasezoey** [comment]
This is somewhat of a known problem, but mongoose's types for `new` and `create` are just *suggestions* and not *required*, only making errors at runtime via the `validate` call.

you can also look at https://github.com/Automattic/mongoose/issues/10302 for some more in-depth discussion, but that issue was not originally about that
